File: app/src/main/java/com/mukeshteckwani/astro/astroapp/model/ChannelsListModel.java
package com.mukeshteckwani.astro.astroapp.model;

import android.databinding.BaseObservable;
import android.databinding.Bindable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import com.mukeshteckwani.astro.astroapp.BR;

import java.util.List;

/**
 * Created by mukeshteckwani on 17/12/17.
 */

public class ChannelsListModel {

    @SerializedName("responseMessage")
    @Expose
    private String responseMessage;
    @SerializedName("responseCode")
    @Expose
    private String responseCode;
    @SerializedName("channels")
    @Expose
    private List<Channel> channels = null;

    public String getResponseMessage() {
        return responseMessage;
    }

    public String getResponseCode() {
        return responseCode;
    }

    public List<Channel> getChannels() {
        return channels;
    }

    public class Channel extends BaseObservable{

        @SerializedName("channelId")
        @Expose
        private Integer channelId;
        @SerializedName("channelTitle")
        @Expose
        private String channelTitle;
        @SerializedName("channelStbNumber")
        @Expose
        private Integer channelStbNumber;

        private boolean isChecked;

        public Integer getChannelId() {
            return channelId;
        }

        public String getChannelTitle() {
            return channelTitle;
        }

        public Integer getChannelStbNumber() {
            return channelStbNumber;
        }

        @Bindable
        public boolean isChecked() {
            return isChecked;
        }

        public void setChecked(boolean checked) {
            isChecked = checked;
            notifyPropertyChanged(BR.checked);
        }
    }
}

File: app/src/main/java/com/mukeshteckwani/astro/astroapp/model/ChannelsListModel.kt
package com.mukeshteckwani.astro.astroapp.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class ChannelsListModel(
    @SerializedName("responseMessage")
    @Expose
    val responseMessage: String? = null,
    @SerializedName("responseCode")
    @Expose
    val responseCode: String? = null,
    @SerializedName("channels")
    @Expose
    val channels: List<Channel>? = null
) {
    data class Channel(
        @SerializedName("channelId")
        @Expose
        val channelId: Int? = null,
        @SerializedName("channelTitle")
        @Expose
        val channelTitle: String? = null,
        @SerializedName("channelStbNumber")
        @Expose
        val channelStbNumber: Int? = null,
        var isChecked: Boolean = false
    ) {
        // Default constructor required for Firebase DataSnapshot.getValue(Channel.class)
        constructor() : this(null, null, null, false)

        override fun equals(other: Any?): Boolean {
            if (this === other) return true
            if (other !is Channel) return false
            return channelId == other.channelId
        }

        override fun hashCode(): Int {
            return channelId ?: 0
        }
    }
}
